src/logging/request-obfuscator.js

```diff
@@ -0,0 +1,133 @@
+
+/**
+ * Module dependencies.
+ */
+
+import _ from 'lodash';
+
+/**
+ * Obfuscating string.
+ */
+
+const obfuscated = '******';
+
+/**
+ * Methods that should be obfuscated.
+ */
+
+const methods = {
+  dumpprivkey: { response: () => obfuscated },
+  encryptwallet: {
+    request: params => {
+      params[0] = obfuscated;
+
+      return params;
+    }
+  },
+  importprivkey: { request: () => [obfuscated] },
+  signmessagewithprivkey: {
+    request: params => {
+      params[0] = obfuscated;
+
+      return params;
+    }
+  },
+  signrawtransaction: {
+    request: params => {
+      params[2] = _.map(params[2], () => obfuscated);
+
+      return params;
+    }
+  },
+  walletpassphrase: {
+    request: params => {
+      params[0] = obfuscated;
+
+      return params;
+    }
+  }
+};
+
+/**
+ * Obfuscate the response body.
+ */
+
+function obfuscateResponseBody(body, methodName) {
+  const method = _.get(methods[methodName], 'response');
+
+  if (!method || _.isEmpty(body.result)) {
+    return body;
+  }
+
+  body.result = method(body.result);
+
+  return body;
+}
+
+/**
+ * Obfuscate the response.
+ */
+
+function obfuscateResponse(request, instance) {
+  const requestBody = JSON.parse(instance.body);
+
+  if (_.isArray(request.response.body)) {
+    const requestMethods = _.map(requestBody, result => result.method);
+
+    request.response.body = _.map(request.response.body, (body, index) => obfuscateResponseBody(body, requestMethods[index]));
+
+    return;
+  }
+
+  request.response.body = obfuscateResponseBody(request.response.body, requestBody.method);
+}
+
+/**
+ * Ofuscate the request body.
+ */
+
+function obfuscateRequestBody(body) {
+  const method = _.get(methods[body.method], 'request');
+
+  if (!method) {
+    return body;
+  }
+
+  body.params = method(body.params);
+
+  return body;
+}
+
+/**
+ * Obfuscate the request.
+ */
+
+function obfuscateRequest(request) {
+  request.body = JSON.parse(request.body);
+
+  if (_.isArray(request.body)) {
+    request.body = _.map(request.body, obfuscateRequestBody);
+  } else {
+    request.body = obfuscateRequestBody(request.body);
+  }
+
+  request.body = JSON.stringify(request.body);
+}
+
+/**
+ * Export `RequestObfuscator`.
+ */
+
+export function obfuscate(request, instance) {
```

Comment:
WDYT about renaming the `instance` argument to something more specific/explicit? Maybe `originatingRequest`?